applications/RANSModellingApplication/custom_strategies/residual_based_bossak_velocity_scheme.h

```diff
@@ -0,0 +1,630 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:   Jordi Cotela
+//                  Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_RESIDUAL_BASED_BOSSAK_VELOCITY_SCHEME_H_INCLUDED)
+#define KRATOS_RESIDUAL_BASED_BOSSAK_VELOCITY_SCHEME_H_INCLUDED
+
+// System includes
+#include <vector>
+
+// External includes
+
+// Project includes
+#include "custom_strategies/relaxed_dof_updater.h"
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "solving_strategies/schemes/scheme.h"
+#include "utilities/time_discretization.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/// A scheme for steady and dynamic equations, using Bossak time integration.
+/**
+ * It can be used for either first- or second-order time derivatives. Elements
+ * and conditions must provide a specialization of SchemeExtension via
+ * their data value container, which allows the scheme to operate independently
+ * of the variable arrangements in the element or condition.
+ */
+template <class TSparseSpace, class TDenseSpace>
+class ResidualBasedBossakVelocityScheme : public Scheme<TSparseSpace, TDenseSpace>
```

Comment:
what is the difference to the one in the core? Cannot you derive from it?

Reply:
This one is a generic one, not coupled to VELOCITY and ACCELERATION variables. I need this to have generic scalar bossak velocity scheme, so I can use the same scheme for different variables